k8s/upgrade/src/bin/upgrade-job/helm/upgrade.rs

```diff
@@ -151,196 +184,217 @@ impl HelmUpgradeBuilder {
         //       mayastor-1.2.3-testing-upgrade-23-35-25-05-2023, mayastor-2.3.0-rc-3
         //    4. The optional group of character(s) mentioned in (3) above, may optionally contain a
         //       '.' followed by a set of numbers. e.g.: mayastor-2.3.4-rc.1, mayastor-2.3.4-alpha.2
-        let core_chart_regex =
+        let core_regex =
             format!(r"^({CORE_CHART_NAME}-[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z0-9]+(\.[0-9]+)?)*)$");
 
-        // Validate if already upgraded for Umbrella chart, and prepare for upgrade for Core chart.
+        // Determine chart variant.
         let chart_variant: HelmChart;
-        let mut core_chart_dir: Option<PathBuf> = None;
-        let mut core_chart_extra_args: Option<Vec<String>> = None;
-        let mut upgrade_values_file: Option<TempFile> = None;
-
-        if Regex::new(umbrella_chart_regex.as_str()) // Case: HelmChart::Umbrella.
-            .context(RegexCompile {
-                expression: umbrella_chart_regex.clone(),
-            })?
-            .is_match(chart.as_str())
-        {
+        if Regex::new(umbrella_regex.as_str())?.is_match(chart.as_ref()) {
```

Comment:
can't you just return umbrella or core from the if else scope?

Reply:
Done, removed it.